run-build-functions.sh

```diff
@@ -669,9 +702,17 @@ install_dependencies() {
 #
 cache_artifacts() {
   echo "Caching artifacts"
+
   cache_cwd_directory ".bundle" "ruby gems"
   cache_cwd_directory "bower_components" "bower components"
-  cache_cwd_directory "node_modules" "node modules"
+
+  if [ "$NETLIFY_YARN_WORKSPACES" == "true" ]
+  then
+    cache_node_modules
+  else
+    cache_cwd_directory "node_modules" "node modules"
+  fi
+
```

Comment:
The idea is to, once we can drop the feature flag, we keep only the `cache_node_modules` function containing all the logic relative to `node_modules` caching.